signac/contrib/project.py

```diff
@@ -1468,7 +1403,7 @@ def temporary_project(self, dir=None):
             yield tmp_project
 
     @classmethod
-    def init_project(cls, *args, root=None, **kwargs):
+    def init_project(cls, *, root=None):
```

Comment:
Do we want to keep the requirement that `root` is a keyword-only argument? All our previous discussions about compatibility layers are somewhat moot now, so I would be willing to revisit that.

It could even be named `path` for consistency with the new terminology. The more I think about it, it'd be really nice to call it `path`. :)

Reply:
Good idea. I do like this. Let's just break everything now and be done with it 🙂

Reply:
Actually I'm going to make this an issue to address in a subsequent PR. This change will be large and will substantially bloat the scope of this PR. It will also require changing functions like `get_project`. We should figure out the full scope and address this in one fell swoop.

Reply:
I created #757 to follow up on this.